create.ps1

```diff
@@ -1,194 +1,254 @@
+
+        if (-Not ($ActionContext.DryRun -eq $True)) {
+            $UserResponse = Invoke-RestMethod @LisaRequest @SplatParams
+
+            $OutputContext.AccountReference = $UserResponse.objectId
+            $Account | Add-Member -NotePropertyName 'password' -NotePropertyValue $UserResponse.temporaryPassword
+        }
+        else {
+            Write-Verbose -Verbose ($Body | ConvertTo-Json)
+
+            $Account | Add-Member -NotePropertyName 'password' -NotePropertyValue "FakePassword"
         }
+
+        #Update the user with all other props
+        $Body = $Account | Select-Object -Property *, 'accountEnabled' -ExcludeProperty @(
+            'changePasswordNextSignIn', 'usageLocation', 'preferredLanguage',
+            'givenName', 'surName', 'displayName', 'userPrincipalName',
+            'password'
+        )
+
+        # Force the account disabled
+        $Body.accountEnabled = $False
+
+        $SplatParams = @{
+            Uri    = "$($config.BaseUrl)/Users/$($OutputContext.AccountReference)/bulk"
+            Method = 'Patch'
+            Body   = $Body
+        }
+
+        if (-Not ($ActionContext.DryRun -eq $True)) {
+            [void] (Invoke-RestMethod @LisaRequest @SplatParams)
+        }
+        else {
+            Write-Verbose -Verbose ($Body | ConvertTo-Json)
+        }
+
+        # Set the manager
+        if ($PersonContext.References.ManagerAccount) {
+            $SplatParams = @{
+                Uri    = "$($Config.BaseUrl)/Users/$($OutputContext.AccountReference)/Manager"
+                Method = 'Put'
+                Body   = $PersonContext.References.ManagerAccount
+            }
+
+            if (-Not ($ActionContext.DryRun -eq $True)) {
+                [void] (Invoke-RestMethod @LisaRequest @SplatParams)
+            }
+
+            Write-Verbose -Verbose "Added Manager $($Manager.displayName) to '$($Person.DisplayName)'"
+        }
+
+        $AuditLogs.Add([PSCustomObject]@{
+                Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                Message = "Created account for '$($Person.DisplayName)'. Id: $($OutputContext.AccountReference)"
+                IsError = $False
+            })
+
+        $OutputContext.Success = $True
     }
 }
+catch {
+    $Exception = $PSItem | Resolve-ErrorMessage
 
-$result = [PSCustomObject]@{
-    Success          = $success
-    Account          = $account
-    AccountReference = $objectId
-    AuditDetails     = $auditMessage
-}
+    Write-Verbose -Verbose $Exception.VerboseErrorMessage
 
-Write-Output $result | ConvertTo-Json -Depth 10
+    $AuditLogs.Add([PSCustomObject]@{
+            Action  = "CreateAccount" # Optionally specify a different action for this audit log
+            Message = "Error creating account [$($Person.DisplayName)]. Error Message: $($Exception.ErrorMessage)."
+            IsError = $True
+        })
+}
```

Comment:
I don´t see how this can be best practice.

The example is also wrong because if there is no auditlog the outputcontext.success will not be set to false

```PowerShell
$auditLogs = [Collections.Generic.List[PSCustomObject]]::new()

$auditLogs.IsError -contains $True
```

the example above will return False, so `$outputContext.Success` will not be updated

Reply:
What I mean is we do the following:

```powershell
$outputContext.Success = $false
try {
    # do logic
    # auditlog iserror = $false
}
catch {
    # upon error
    # auditlog iserror = $true
}
finally {
   # Check if auditLogs contains errors, if no errors are found, set success to true
    if (!($outputContext.AuditLogs.IsError -contains $true)) {
        $outputContext.Success = $true
    } 
}
``` 
This way the auditlogs are always filled